agent/dns.go

```diff
@@ -1286,8 +1295,9 @@ func trimUDPResponse(req, resp *dns.Msg, udpAnswerLimit int) (trimmed bool) {
 	// will allow our responses to be compliant even if some downstream server
 	// uncompresses them.
 	// Even when size is too big for one single record, try to send it anyway
-	// (useful for 512 bytes messages)
-	for len(resp.Answer) > 1 && resp.Len() > maxSize-7 {
+	// (useful for 512 bytes messages). 8 is removed from maxSize to ensure that we account
+	// for the udp header (8 bytes).
+	for len(resp.Answer) > 1 && resp.Len() > maxSize-8 {
```

Comment:
Chatted offline about this but it should be fine. Not all maxSize will be set to the maxUDPDatagramSize so we account for that remaining 8 bytes of header here. Additionally the binary search trimming also introduces fuzziness so this shouldn't be something we ever run into. Correcting 7 bytes to 8 bytes since that's the actual size of the header.